src/main/java/sg/dex/starfish/keeper/DirectPurchaseAdapter.java

```diff
@@ -0,0 +1,111 @@
+        boolean tokenApproved = false;
+        TransactionReceipt receipt = null;
+        try {
+            tokenApproved = tokenApprove(directPurchase.getContractAddress(), amount.toString());
+        } catch (TokenApproveException e) {
+            e.printStackTrace();
+            return receipt;
+        }
+
+        if(tokenApproved) {
+            try {
+                byte[] ref1 = Numeric.hexStringToByteArray(Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(reference1), 64));
+                byte[] ref2 = Numeric.hexStringToByteArray(Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(reference2), 64));
+                receipt = directPurchase.sendTokenAndLog(to, amount, ref1, ref2).send();
+            } catch (Exception e) {
+                e.printStackTrace();
+                return receipt;
+            }
+            return receipt;
+        }
+        return receipt;
+    }
+
+    public boolean tokenApprove(String spenderAddress, String price) throws TokenApproveException {
+        return tokenManager.tokenApprove(spenderAddress, price);
+    }
+
+    public boolean checkIsPaid(String purchaser_address, String publisher_address, BigInteger amount, String reference)
+    {
+        String purchaser_padded = Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(purchaser_address), 64);
+        String publisher_padded = Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(publisher_address), 64);
+        String reference_padded = Numeric.toHexStringWithPrefixZeroPadded(Numeric.toBigInt(reference), 64);
+        EthFilter filter = new EthFilter(DefaultBlockParameter.valueOf(BigInteger.valueOf(1)), DefaultBlockParameterName.LATEST, directPurchase.getContractAddress());
+
+        filter.addSingleTopic(EventEncoder.encode(directPurchase.TOKENSENT_EVENT));
+        // addOtionalTopic does not work for some reason.
+        filter.addSingleTopic(purchaser_padded);
+        filter.addSingleTopic(publisher_padded);
+        filter.addSingleTopic(reference_padded);
+
+        Flowable<DirectPurchase.TokenSentEventResponse> floable = directPurchase.tokenSentEventFlowable(filter);
+        ArrayList<DirectPurchase.TokenSentEventResponse> outcome = new ArrayList<>();
+        floable.subscribe(log -> {
+            outcome.add(log);
+        });
+
+        for (DirectPurchase.TokenSentEventResponse obj:outcome) {
+            if (obj._amount.equals(amount))
+                return true;
+        }
+        return false;
+    }
+
+    private Properties getProperties() {
+        Properties prop = new Properties();
+        try (InputStream input = getClass().getClassLoader().getResourceAsStream("application_spree.properties")) {
+
+            if (input == null) {
+                throw new IOException("properties files is missing");
+            }
+
+            prop.load(input);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Why are we consuming the exception here?

Have we actually recovered? If not then should re-throw the exception. (logging is fine of course)

Reply:
whole private Properties getProperties() everywhere in the project will be reviewed and probably remade somehow @AyushVerma2